### PR TITLE
Add support for track parameter for ucscTableQuery

### PR DESCRIPTION
**Pull request by sanchit-saini**
Fix for #36

### CODE REVIEW

File: R/ucsc.R [hunk 1]
```diff
@@ -270,19 +270,28 @@ setMethod("ucscTableQuery", "character",
...
                    genome = NULL, url = "http://genome.ucsc.edu/cgi-bin/") {
               stopifnot(isSingleString(x))
-              if (!isSingleString(table))
-                stop("'table' is a mandatory parameter and must be a single character vector")
               if (!is.null(intersectTrack))
                 stop("intersectTrack is no longer supported")
-              if (!is.null(track))
-                warning("track is meaningless now you only go by the table")
+              # if (!is.null(track))
+              #   warning("track is meaningless now you only go by the table")
```
Comment (lawremi) [inline R/ucsc.R:276]: We will need to issue a deprecation warning so that packages know to start updating.

File: R/ucsc.R [hunk 2]
```diff
@@ -270,19 +270,28 @@ setMethod("ucscTableQuery", "character",
...
-                  stop("'genome' is a mandatory parameter and must be a single character vector")
+                  stop("'genome' is a mandatory parameter and must be a single string")
                 hubUrl <- x
               } else genome <- x
+              if (!is.null(track)) {
+                trackids <- ucscTableTracks(genome)
+                if (track %in% names(trackids)) {
+                  track <- trackids[[track]]
+                } else stop("Unknown track: ", track)
+                form <- c(db = genome, hgta_group = "allTracks", hgta_track = track)
```
Comment (lawremi) [inline R/ucsc.R:289]: It would be cleaner to separate the HTML-level logic into a separate function.

File: R/ucsc.R [hunk 3]
```diff
@@ -457,68 +509,114 @@ setMethod("track", "UCSCTableQuery",
...
+  columnTypes <- response[["columnTypes"]]
+  columnNames <- vapply(columnTypes, function(x) x$name, character(1L))
+  tableName <- response[["track"]]
+  results <- response[[tableName]]
+  df <- listToDataframe(results, columnNames)
+  df[["tracktype"]] <- "bigBed"
+  df
+}
+
+parseBwDf <- function(response) {
```
Comment (lawremi) [inline R/ucsc.R:574]: Please rename to `parseBigWigDf` and `parseBigBedDf`.

File: NAMESPACE [hunk 4]
```diff
@@ -119,7 +119,7 @@ exportMethods(activeView, "activeView<-", blocks, browseGenome,
 export(genomeBrowsers, start, end, strand, "start<-", "end<-", width, "width<-",
        ranges, values, GenomicData, GenomicSelection,
        score, "score<-", as.data.frame, space,
-       ucscGenomes,
+       ucscGenomes, ucscTables,
```
Comment (lawremi) [inline NAMESPACE:122]: Please document `ucscTables`

Comment (sanchit-saini) [reply NAMESPACE:122]: In which file should I add the documentation for it?
As  `ucscTables` is not part of any class.

Comment (sanchit-saini) [reply NAMESPACE:122]: For now, I added it under the `UCSCTableQuery` as `ucscTables` is related to the table query interface.

**Pull request merged**